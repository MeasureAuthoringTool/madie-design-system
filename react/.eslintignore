_*
css
dist
node_modules
test
