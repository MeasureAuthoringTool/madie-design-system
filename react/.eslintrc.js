module.exports = {
    extends: ['eslint:recommended', 'plugin:react/recommended'],
    parser: 'babel-eslint',
    parserOptions: {
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true
        }
    },
    env: {
        es6: true,
        node: true,
        browser: true,
        jquery: true,
        mocha: true
    },
    rules: {
        'no-unused-vars': [
            'error',
            {
                args: 'none'
            }
        ],
        'linebreak-style': ['error', 'unix'],
        'no-console': 0,
        'no-prototype-builtins': [0]
    },
    settings: {
        react: { version: 'detect' }
    }
};
