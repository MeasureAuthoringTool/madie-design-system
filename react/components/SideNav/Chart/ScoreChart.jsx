import React, { useState, useEffect } from "react";
import { pie, arc } from "d3-shape";
import { select, event } from "d3-selection";
import PropTypes from "prop-types";

const WIDTH = 210;
const HEIGHT = 210;
const RADIUS = Math.min(WIDTH, HEIGHT) / 2;
const GAP_VALUE = 2.5;

const ScoreChart = ({ performanceYear, chartData, linkCallback }) => {
    const [hasLegend, setHasLegend] = useState(false);
    const [titleMapping, setTitleMapping] = useState({
        quality: "Quality Measures",
        aci: "Advancing Care Info",
        pi: "Promoting Interoperability",
        ia: "Improvement Activities",
        cost: "Cost",
    });

    const { categories } = chartData;
    let legendToggler;
    let legend;

    const getAdvancingCareLabel = () => {
        if (performanceYear > 2017) {
            setTitleMapping({
                ...titleMapping,
                aci: "Promoting Interoperability",
            });
        }
    };

    const redirectToCategory = (link, categoryId) => {
        if (!link) {
            return;
        }

        // track explicitly chart legends
        if (window.utag) {
            window.utag.track("link", {
                ga_event_category: "SidebarNav",
                ga_event_action: `GoTo${categoryId}`,
                ga_event_label: categoryId,
            });
        }

        if (linkCallback) {
            linkCallback(link);
        } else {
            window.location.href = link;
        }
    };

    const draw = (chartData) => {
        const scorePie = pie().sort(null);

        // Removing invalid categories
        const categories = chartData.categories.filter(
            ({ maxContribution }) => !!maxContribution
        );

        // Calculating the sum of valid categories in order to calculate ratios
        const categorySum = categories
            .map(({ maxContribution }) => maxContribution)
            .reduce((sum, current) => sum + current);

        if (!categorySum) {
            return;
        }

        select(".score-chart").selectAll("*").remove();

        const svg = select(".score-chart")
            .attr("width", WIDTH)
            .attr("height", HEIGHT);

        const chartArc = arc()
            .innerRadius(RADIUS - 1.5)
            .outerRadius(RADIUS - 25);

        // Preparing the data structure with ratios: [ FILLED, EMPTY, TRANSPARENT, FILLED, EMPTY, ... ]
        let data;
        if (categories.length === 0) {
            data = [];
        } else if (categories.length === 1) {
            const { value, maxContribution } = categories[0];
            data = [value || 0, maxContribution - value];
        } else {
            data = categories
                .map(({ value, maxContribution }) => [
                    value || 0,
                    maxContribution - value,
                    GAP_VALUE * (categorySum / 100),
                ])
                .reduce((a, b) => a.concat(b), []);
        }

        svg.append("title").attr("id", "titleID").text("Your submission score");

        svg.append("desc")
            .attr("id", "descID")
            .text(`${chartData.finalScore} out of 100`);

        // Setting up the arcs
        let arcs = svg
            .append("g")
            .attr("class", "arcs")
            .attr("role", "presentation")
            .selectAll("g.arc")
            .data(scorePie(data))
            .enter()
            .append("g")
            .attr("class", (d, i) => {
                let role;
                if (i % 3 === 0) {
                    role = "filled";
                } else if (i % 3 === 1) {
                    role = "empty";
                } else {
                    role = "transparent";
                }
                return `arc ${categories[parseInt(i / 3)].name} ${role}`;
            })
            .attr("transform", `translate(${RADIUS}, ${RADIUS})`)
            // For accessibility, we add keyboard access to the filled slices
            .attr("tabindex", (d, i) => {
                let tabindex = -1;
                if (i % 3 === 0) {
                    tabindex = 0;
                }
                return tabindex;
            })
            // IE for for focusable svg elements
            .attr("focusable", (d, i) => {
                if (i % 3 === 0) {
                    return true;
                }
            })
            // For accessibility, we add keyboard access to the filled slices
            .on("keyup", (d, i) => {
                const code = event.keyCode || event.which;
                if (code === 13) {
                    const { name, link } = categories[parseInt(i / 3)];
                    redirectToCategory(link, name);
                }
            })
            // On click - go to category page
            .on("click", (d, i) => {
                const { name, link } = categories[parseInt(i / 3)];
                redirectToCategory(link, name);
            });

        // Add accessibility description
        arcs.append("desc").text((d, i) => {
            if (i % 3 === 0) {
                const { name, value, maxContribution } =
                    categories[parseInt(i / 3)];
                return `${titleMapping[name]} ${value} out of ${maxContribution}`;
            }
        });

        arcs.append("path")
            .attr("d", chartArc)
            // On mouse over - show category tooltip
            .on("mouseover", (d, i) => {
                const { name, value, maxContribution } =
                    categories[parseInt(i / 3)];
                select(".chart-tooltip")
                    .style("opacity", 1)
                    .text(`${titleMapping[name]} ${value}/${maxContribution}`);
            })
            // On mouse move - move category tooltip
            .on("mousemove", () => {
                select(".chart-tooltip")
                    .style("top", event.clientY - HEIGHT + 10 + "px")
                    .style("left", event.clientX + "px");
            })
            // On mouse out - hide category tooltip
            .on("mouseout", () => {
                select(".chart-tooltip").style("opacity", 0);
            });

        // Adding title and subtitle
        const text = svg
            .append("g")
            .attr("text-anchor", "middle")
            .attr("transform", `translate(${WIDTH / 2}, ${HEIGHT / 2})`);

        // Adding title - X%
        text.append("text")
            .attr("y", 4)
            .attr("class", "chart-title")
            .text(chartData.finalScore);

        // Adding subtitle - OUT OF 100
        text.append("text")
            .attr("y", 24)
            .attr("class", "chart-subtitle")
            .text("OUT OF 100");
    };

    const toggleLegend = () => {
        setHasLegend(!hasLegend);
    };

    useEffect(() => {
        if (chartData?.finalScore) {
            draw(chartData);
            getAdvancingCareLabel();
        }
    }, [chartData?.finalScore]);

    if (hasLegend) {
        legendToggler = (
            <button className="open" type="button" aria-pressed="true">
                Hide Legend
                <svg
                    className="right-icon rotated"
                    aria-hidden="true"
                    focusable="false"
                >
                    <use xlinkHref="#chevron-down" />
                </svg>
            </button>
        );
        legend = categories.map(
            ({ name, value, maxContribution, link }, index) => {
                return (
                    <li key={name} className="legend-axis">
                        <button onClick={() => redirectToCategory(link, name)}>
                            <span className={`legend-axis-color ${name}`} />
                            &nbsp;
                            <span className="legend-axis-title">
                                {titleMapping[name]}
                            </span>
                            &nbsp;
                            <span className="legend-axis-value">{`${value}/${maxContribution}`}</span>
                        </button>
                    </li>
                );
            }
        );
    } else {
        legendToggler = (
            <button type="button" aria-pressed="false">
                Show Legend
                <svg
                    className="right-icon"
                    aria-hidden="true"
                    focusable="false"
                >
                    <use xlinkHref="#chevron-down" />
                </svg>
            </button>
        );
    }

    return (
        <div className="chart">
            <svg className="score-chart" focusable="false" />
            <div className="chart-tooltip off" />
            <div className="chart-legend-toggler" onClick={toggleLegend}>
                {legendToggler}
            </div>
            <ul className="chart-legend">{legend}</ul>
        </div>
    );
};

ScoreChart.propTypes = {
    chartData: PropTypes.object,
    linkCallback: PropTypes.func,
    performanceYear: PropTypes.number,
};

export default ScoreChart;
