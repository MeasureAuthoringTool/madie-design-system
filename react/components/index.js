import Accordion from "./Accordion";
import Alert from "./Alert";
import AutoComplete from "./AutoComplete/AutoComplete";
import Breadcrumb from "./Breadcrumb";
import Button, { TextButton } from "./Button";
import DateField from "./DateField/DateField";
import DateTimeField from "./DateTimeField/DateTimeField";
import FlashNotification from "./FlashNotification/FlashNotificationUI";
import Instant from "./Instant";
import InputLabel from "./InputLabel";
import FormControlLabel from "./FormControlLabel";
import MadieAlert from "./MadieAlert";
import MadieDialog from "./MadieDialog";
import MadieDiscardDialog from "./MadieDiscardDialog";
import MadieDeleteDialog from "./MadieDeleteDialog";
import MadieConfirmDialog from "./MadieConfirmDialog";
import MadieSpinner from "./MadieSpinner";
import MadieTooltip from "./MadieTooltip";
import MadieTooltipIcon from "./MadieTooltipIcon";
import Modal from "./Modal";
import Popover from "./Popover";
import { Pagination } from "./Pagination";
import RadioButton from "./RadioButton/RadioButton";
import ReadOnlyTextField from "./ReadOnlyTextField";
import RichTextEditor from "./RichTextEditor";
import Select from "./Select";
import Spinner from "./UnwrappedSpinner";
import TabPanel from "./Tabs/TabPanel";
import Tabs from "./Tabs/index";
import Tab from "./Tabs/Tab";
import TextField from "./TextField";
import TimeField from "./TimeField/TimeField";
import TextArea from "./TextArea";
import Toast from "./Toast";
import Search from "./Search";
import TextInput from "./TextInput";
import Dropdown from "./Dropdown";
import DSLink from "./Link";
import NumberInput from "./NumberInput";
import TruncateText from "./TruncateText/index";
import theme from "../themes/actionCenterTheme.js";
import {
    MyApplicationsIcon,
    UserSignInIcon,
    MyTestDataIcon,
    ReportDataIcon,
    ManageUsersIcon,
    AccountHomeIcon,
    FacilityBasedPreviewIcon,
    PaymentIcon,
    PhysicianCompareIcon,
    HardshipIcon,
    TargetIcon,
    ReportsIcon,
    HelpSupportIcon,
    DashboardIcon,
    AccountSettings,
    DetailsIcon,
    StarIcon,
    CliniciansIcon,
    GroupReporting,
    IndividualReporting,
    ChevronLeft,
    ChevronRight,
    ChevronDown,
    SwitchPractice,
    ScoreIncrease,
    TooltipIcon,
    NavCollapse,
    NavExpand,
    CloseXIcon,
    CloseXIcon2,
    BellOutline,
    EarlyClaims,
    CliniciansList,
    YoutubeIcon,
    TwitterIcon,
    WiViewReportsIcon,
    WiViewProgressIcon,
    WiManageGroupIcon,
} from "../lib/SvgComponents.jsx";

const Icons = {
    MyApplicationsIcon,
    UserSignInIcon,
    MyTestDataIcon,
    ReportDataIcon,
    ManageUsersIcon,
    AccountHomeIcon,
    FacilityBasedPreviewIcon,
    PaymentIcon,
    PhysicianCompareIcon,
    HardshipIcon,
    TargetIcon,
    ReportsIcon,
    HelpSupportIcon,
    DashboardIcon,
    AccountSettings,
    DetailsIcon,
    StarIcon,
    CliniciansIcon,
    GroupReporting,
    IndividualReporting,
    ChevronLeft,
    ChevronRight,
    ChevronDown,
    SwitchPractice,
    ScoreIncrease,
    TooltipIcon,
    NavCollapse,
    NavExpand,
    CloseXIcon,
    CloseXIcon2,
    BellOutline,
    EarlyClaims,
    CliniciansList,
    YoutubeIcon,
    TwitterIcon,
    WiViewReportsIcon,
    WiViewProgressIcon,
    WiManageGroupIcon,
};

export {
    Accordion,
    Alert,
    AutoComplete,
    Breadcrumb,
    Button,
    DateField,
    DateTimeField,
    FlashNotification,
    FormControlLabel,
    InputLabel,
    Icons,
    Instant,
    Modal,
    MadieAlert,
    MadieDialog,
    MadieDiscardDialog,
    MadieDeleteDialog,
    MadieConfirmDialog,
    MadieSpinner,
    MadieTooltipIcon,
    MadieTooltip,
    NumberInput,
    Pagination,
    Popover,
    RadioButton,
    ReadOnlyTextField,
    RichTextEditor,
    Search,
    Select,
    Spinner,
    TabPanel,
    Tabs,
    Tab,
    TextButton,
    TextInput,
    TimeField,
    TextField,
    TextArea,
    Toast,
    Dropdown,
    DSLink,
    TruncateText,
    theme,
};
