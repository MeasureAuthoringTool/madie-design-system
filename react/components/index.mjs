
import Accordion from "./Accordion/index.jsx";
import AutoComplete from "./AutoComplete/AutoComplete.jsx";
import Button, { TextButton } from "./Button/index.js";
import DateField from "./DateField/DateField.jsx";
import DateTimeField from "./DateTimeField/DateTimeField.jsx";
import DSLink from "./Link/index.js";
import Instant from "./Instant/index.jsx";
import InputLabel from "./InputLabel/index.jsx";
import FormControlLabel from "./FormControlLabel/index.jsx";
import MadieAlert from "./MadieAlert/index.jsx";
import MadieDialog from "./MadieDialog/index.jsx";
import MadieDiscardDialog from "./MadieDiscardDialog/index.jsx";
import MadieDeleteDialog from "./MadieDeleteDialog/index.jsx";
import MadieConfirmDialog from "./MadieConfirmDialog/index.jsx";
import MadieSpinner from "./MadieSpinner/index.jsx";
import MadieTooltip from "./MadieTooltip/index.jsx";
import MadieTooltipIcon from "./MadieTooltipIcon/index.jsx";
import Modal from "./Modal/index.jsx";
import Popover from "./Popover/index.js";
import { Pagination } from "./Pagination/index.jsx";
import RadioButton from "./RadioButton/RadioButton.jsx";
import ReadOnlyTextField from "./ReadOnlyTextField/index.jsx";
import RichTextEditor from "./RichTextEditor/index.jsx";
import Select from "./Select/index.jsx";
import TabPanel from "./Tabs/TabPanel.js";
import Tabs from "./Tabs/index.js";
import Tab from "./Tabs/Tab.js";
import TextField from "./TextField/index.jsx";
import TimeField from "./TimeField/TimeField.jsx";
import TextArea from "./TextArea/index.jsx";
import Toast from "./Toast/index.jsx";
import Search from "./Search/index.js";
import NumberInput from "./NumberInput/index.jsx";
import TruncateText from "./TruncateText/index.js";
import theme from "../themes/actionCenterTheme.js";

import {
    MyApplicationsIcon,
    UserSignInIcon,
    MyTestDataIcon,
    ReportDataIcon,
    ManageUsersIcon,
    AccountHomeIcon,
    FacilityBasedPreviewIcon,
    PaymentIcon,
    PhysicianCompareIcon,
    HardshipIcon,
    TargetIcon,
    ReportsIcon,
    HelpSupportIcon,
    DashboardIcon,
    AccountSettings,
    DetailsIcon,
    StarIcon,
    CliniciansIcon,
    GroupReporting,
    IndividualReporting,
    ChevronLeft,
    ChevronRight,
    ChevronDown,
    SwitchPractice,
    ScoreIncrease,
    TooltipIcon,
    NavCollapse,
    NavExpand,
    CloseXIcon,
    CloseXIcon2,
    BellOutline,
    EarlyClaims,
    CliniciansList,
    YoutubeIcon,
    TwitterIcon,
    WiViewReportsIcon,
    WiViewProgressIcon,
    WiManageGroupIcon,
} from "../lib/SvgComponents.jsx";

const Icons = {
    MyApplicationsIcon,
    UserSignInIcon,
    MyTestDataIcon,
    ReportDataIcon,
    ManageUsersIcon,
    AccountHomeIcon,
    FacilityBasedPreviewIcon,
    PaymentIcon,
    PhysicianCompareIcon,
    HardshipIcon,
    TargetIcon,
    ReportsIcon,
    HelpSupportIcon,
    DashboardIcon,
    AccountSettings,
    DetailsIcon,
    StarIcon,
    CliniciansIcon,
    GroupReporting,
    IndividualReporting,
    ChevronLeft,
    ChevronRight,
    ChevronDown,
    SwitchPractice,
    ScoreIncrease,
    TooltipIcon,
    NavCollapse,
    NavExpand,
    CloseXIcon,
    CloseXIcon2,
    BellOutline,
    EarlyClaims,
    CliniciansList,
    YoutubeIcon,
    TwitterIcon,
    WiViewReportsIcon,
    WiViewProgressIcon,
    WiManageGroupIcon,
};

export {
    Accordion,
    AutoComplete,
    Button,
    DateField,
    DateTimeField,
    DSLink,
    FormControlLabel,
    InputLabel,
    Icons,
    Instant,
    Modal,
    MadieAlert,
    MadieDialog,
    MadieDiscardDialog,
    MadieDeleteDialog,
    MadieConfirmDialog,
    MadieSpinner,
    MadieTooltipIcon,
    MadieTooltip,
    NumberInput,
    Pagination,
    Popover,
    RadioButton,
    ReadOnlyTextField,
    RichTextEditor,
    Search,
    Select,
    TabPanel,
    Tabs,
    Tab,
    TextButton,
    TimeField,
    TextField,
    TextArea,
    Toast,
    TruncateText,
    theme,
};
