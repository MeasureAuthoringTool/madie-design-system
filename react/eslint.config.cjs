const {
    defineConfig,
    globalIgnores,
} = require("eslint/config");

const babelParser = require("@babel/eslint-parser");
const globals = require("globals");
const js = require("@eslint/js");

const {
    FlatCompat,
} = require("@eslint/eslintrc");

const compat = new FlatCompat({
    baseDirectory: __dirname,
    recommendedConfig: js.configs.recommended,
    allConfig: js.configs.all
});

module.exports = defineConfig([{
    extends: compat.extends("eslint:recommended", "plugin:react/recommended"),

    languageOptions: {
        parser: babelParser,
        sourceType: "module",

        parserOptions: {
            ecmaFeatures: {
                jsx: true,
            },
            ecmaVersion: "latest",
        },

        globals: {
            ...globals.node,
            ...globals.browser,
            ...globals.jquery,
            ...globals.mocha,
        },
    },

    rules: {
        "no-unused-vars": ["error", {
            args: "none",
        }],

        "linebreak-style": ["error", "unix"],
        "no-console": 0,
        "no-prototype-builtins": [0],
    },

    settings: {
        react: {
            version: "detect",
        },
    },
}, globalIgnores(["**/_*", "**/css", "**/dist", "**/node_modules", "**/test"])]);
