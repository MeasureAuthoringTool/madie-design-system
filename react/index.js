import ErrorPage from './components/Error/error';
import Footer from './components/Footer/footer';
import Header from './components/Header/header';
import SideNav from './components/SideNav';
import SessionDialog from './components/SessionDialog/sessionDialog';

import 'what-input';

const QppStyleComponents = {
  errorPage(options) {
    return new ErrorPage(options);
  },

  footer(options) {
    return new Footer(options);
  },

  header(options) {
    return new Header(options);
  },

  sideNav(options) {
    return SideNav(options);
  },

  SessionDialog(options) {
    return new SessionDialog(options);
  },
};

export default QppStyleComponents;
