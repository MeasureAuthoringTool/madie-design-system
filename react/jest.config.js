module.exports = {
    roots: ["<rootDir>"],
    modulePathIgnorePatterns: ["<rootDir>/dist/"],
    testEnvironment: "jsdom",
    transform: {
        "^.+\\.(j|t)sx?$": "babel-jest",
    },
    setupFilesAfterEnv: ["@testing-library/jest-dom"],
    globals: {
        "ts-jest": {
            tsconfig: {
                jsx: "react-jsx",
            },
            useEsm: true,
        },
    },
};
