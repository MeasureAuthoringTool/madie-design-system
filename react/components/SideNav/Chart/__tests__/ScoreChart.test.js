/* global expect */
import React from "react";
import { render } from "@testing-library/react";
import ScoreChart from "../ScoreChart";

const CHART_DATA = {
    finalScore: 50,
    submissionId: "x",
    categories: [
        { name: "quality", value: 30, maxContribution: 60 },
        { name: "aci", value: 15, maxContribution: 30 },
        { name: "ia", value: 5, maxContribution: 10 },
    ],
};

describe("ScoreChart component", () => {
    it("should match snapshot", () => {
        const { container } = render(<ScoreChart chartData={CHART_DATA} />);

        expect(container).toMatchSnapshot();
    });
});
