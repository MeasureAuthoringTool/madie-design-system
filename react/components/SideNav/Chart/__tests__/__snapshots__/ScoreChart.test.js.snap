// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreChart component should match snapshot 1`] = `
<div>
  <div
    class="chart"
  >
    <svg
      class="score-chart"
      focusable="false"
      height="210"
      width="210"
    >
      <title
        id="titleID"
      >
        Your submission score
      </title>
      <desc
        id="descID"
      >
        50 out of 100
      </desc>
      <g
        class="arcs"
        role="presentation"
      >
        <g
          class="arc quality filled"
          focusable="true"
          tabindex="0"
          transform="translate(105, 105)"
        >
          <desc>
            Quality Measures 30 out of 60
          </desc>
          <path
            d="M6.3375471855875526e-15,-103.5A103.5,103.5,0,0,1,101.77834755184973,18.799414076371672L78.66925414635729,14.530948078354914A80,80,0,0,0,4.898587196589413e-15,-80Z"
          />
        </g>
        <g
          class="arc quality empty"
          tabindex="-1"
          transform="translate(105, 105)"
        >
          <desc />
          <path
            d="M101.77834755184973,18.799414076371672A103.5,103.5,0,0,1,-36.97339709441724,96.67066725381872L-28.578471184090617,74.72128869860384A80,80,0,0,0,78.66925414635729,14.530948078354914Z"
          />
        </g>
        <g
          class="arc quality transparent"
          tabindex="-1"
          transform="translate(105, 105)"
        >
          <desc />
          <path
            d="M-36.97339709441724,96.67066725381872A103.5,103.5,0,0,1,-50.65474724844733,90.2571137428841L-39.15342782488682,69.76395265150461A80,80,0,0,0,-28.578471184090617,74.72128869860384Z"
          />
        </g>
        <g
          class="arc aci filled"
          focusable="true"
          tabindex="0"
          transform="translate(105, 105)"
        >
          <desc>
            Advancing Care Info 15 out of 30
          </desc>
          <path
            d="M-50.65474724844733,90.2571137428841A103.5,103.5,0,0,1,-101.77834755184973,18.79941407637173L-78.66925414635729,14.53094807835496A80,80,0,0,0,-39.15342782488682,69.76395265150461Z"
          />
        </g>
        <g
          class="arc aci empty"
          tabindex="-1"
          transform="translate(105, 105)"
        >
          <desc />
          <path
            d="M-101.77834755184973,18.79941407637173A103.5,103.5,0,0,1,-79.55497896707813,-66.20615773134517L-61.491771182282605,-51.17384172471125A80,80,0,0,0,-78.66925414635729,14.53094807835496Z"
          />
        </g>
        <g
          class="arc aci transparent"
          tabindex="-1"
          transform="translate(105, 105)"
        >
          <desc />
          <path
            d="M-79.55497896707813,-66.20615773134517A103.5,103.5,0,0,1,-69.06749696150186,-77.08392091398136L-53.385504897779214,-59.5817746195025A80,80,0,0,0,-61.491771182282605,-51.17384172471125Z"
          />
        </g>
        <g
          class="arc ia filled"
          focusable="true"
          tabindex="0"
          transform="translate(105, 105)"
        >
          <desc>
            Improvement Activities 5 out of 10
          </desc>
          <path
            d="M-69.06749696150186,-77.08392091398136A103.5,103.5,0,0,1,-43.93126833364699,-93.71389258054062L-33.956535910065305,-72.43585899945168A80,80,0,0,0,-53.385504897779214,-59.5817746195025Z"
          />
        </g>
        <g
          class="arc ia empty"
          tabindex="-1"
          transform="translate(105, 105)"
        >
          <desc />
          <path
            d="M-43.93126833364699,-93.71389258054062A103.5,103.5,0,0,1,-15.069720860573184,-102.39703859577388L-11.64809341880053,-79.1474694460088A80,80,0,0,0,-33.956535910065305,-72.43585899945168Z"
          />
        </g>
        <g
          class="arc ia transparent"
          tabindex="-1"
          transform="translate(105, 105)"
        >
          <desc />
          <path
            d="M-15.069720860573184,-102.39703859577388A103.5,103.5,0,0,1,-2.0286557443468859e-13,-103.5L-1.5680430874178828e-13,-80A80,80,0,0,0,-11.64809341880053,-79.1474694460088Z"
          />
        </g>
      </g>
      <g
        text-anchor="middle"
        transform="translate(105, 105)"
      >
        <text
          class="chart-title"
          y="4"
        >
          50
        </text>
        <text
          class="chart-subtitle"
          y="24"
        >
          OUT OF 100
        </text>
      </g>
    </svg>
    <div
      class="chart-tooltip off"
    />
    <div
      class="chart-legend-toggler"
    >
      <button
        aria-pressed="false"
        type="button"
      >
        Show Legend
        <svg
          aria-hidden="true"
          class="right-icon"
          focusable="false"
        >
          <use
            xlink:href="#chevron-down"
          />
        </svg>
      </button>
    </div>
    <ul
      class="chart-legend"
    />
  </div>
</div>
`;
